src/data/list/duplicate.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2018 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky, Chris Hughes
+-/
+import data.list.basic
+
+/-!
+# List duplicates
+
+## Main definitions
+
+* `list.duplicate x l : Prop` is an inductive property that holds when `x` is a duplicate in `l`
+
+-/
+
+variable {α : Type*}
+
+namespace list
+
+/-- Property that an element `x : α` of `l : list α` can be found in the list more than once. -/
+inductive duplicate (x : α) : list α → Prop
+| cons_mem {l : list α} : x ∈ l → duplicate (x :: l)
+| cons_duplicate {y : α} {l : list α} : duplicate l → duplicate (y :: l)
+
+local infix ` ∈+ `:1 := list.duplicate
```

Comment:
This is the precedence of `∈`. This should allow you to remove parentheses around this notation (like in `duplicate_cons_self_iff`)
```suggestion
local infix ` ∈+ `:50 := list.duplicate
```